tests/integrations/raftstore/test_multi.rs

```diff
@@ -791,3 +791,33 @@ fn test_server_batch_write() {
     let mut cluster = new_server_cluster(0, 3);
     test_batch_write(&mut cluster);
 }
+
+// Tests whether logs are catch up quickly.
+#[test]
+fn test_node_catch_up_logs() {
+    let mut cluster = new_node_cluster(0, 3);
+    cluster.cfg.raft_store.raft_base_tick_interval = ReadableDuration::millis(500);
+    cluster.cfg.raft_store.raft_max_size_per_msg = ReadableSize(5);
+    cluster.cfg.raft_store.raft_election_timeout_ticks = 50;
+    cluster.cfg.raft_store.max_leader_missing_duration = ReadableDuration::hours(1);
+    cluster.cfg.raft_store.peer_stale_state_check_interval = ReadableDuration::minutes(30);
+    cluster.cfg.raft_store.abnormal_leader_missing_duration = ReadableDuration::hours(1);
+    // disable compact log to make test more stable.
+    cluster.cfg.raft_store.raft_log_gc_threshold = 3000;
+    cluster.pd_client.disable_default_operator();
+    // We use three peers([1, 2, 3]) for this test.
+    let r1 = cluster.run_conf_change();
+    cluster.pd_client.must_add_peer(r1, new_peer(2, 2));
+    cluster.pd_client.must_add_peer(r1, new_peer(3, 3));
+
+    cluster.must_put(b"k1", b"v1");
+    must_get_equal(&cluster.get_engine(3), b"k1", b"v1");
+    cluster.stop_node(3);
+    for i in 0..10 {
+        let v = format!("{:04}", i);
+        cluster.async_put(v.as_bytes(), v.as_bytes()).unwrap();
+    }
+    must_get_equal(&cluster.get_engine(1), b"0009", b"0009");
+    cluster.run_node(3).unwrap();
+    must_get_equal(&cluster.get_engine(3), b"0009", b"0009");
```

Comment:
There are only 10 logs. Does it really test the speed of catching up logs?

Reply:
Because `size_per_message` is set to 5, so each message will only contains 1 entry. Node 3 will requires at least 10 messages to catch up logs. Without aggressive flow control, the total time will be more than 500ms * 10 = 5 seconds in slow tests run time, hence the line will timeout.